### PR TITLE
Add ChefZero::Dist to abstract branding details to a single location

### PR DESCRIPTION
**Pull request by Tensibai**
Related to chef/chef's [#8376](https://github.com/chef/chef/issues/8376) design proposal to set brands in a single location for easy patching by community distributions.

## Description
- Add a dist.rb file defining the PRODUCT and CLIENT constants to be used in outputs.
- Replace occurences of chef-zero and Chef-Zero by their constant counterparts.
- add a require "chef_zero/dist.rb" to files updated on point above

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef_zero/rspec.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 require "tempfile"
-require "chef_zero/server"
+require "chef_zero/ser1ver"
```
Comment (stevendanna) [inline lib/chef_zero/rspec.rb]:
```suggestion
require "chef_zero/server"
```

Comment (Tensibai) [reply lib/chef_zero/rspec.rb]: Oops, typo somewhere when adding the line, sorry about that :(

**Comment by Tensibai** [comment]
@stevendanna do I need to redo the last change on my side and force push to sign it to avoid the check problem or could it be left as-is ?

File: bin/chef-zero [hunk 2]
```diff
@@ -100,7 +101,7 @@ if options[:daemon]
     server.start(true)
   else
     if ENV["OS"] == "Windows_NT"
-      abort "Daemonization is not supported on Windows. Running 'start chef-zero' will fork the process."
+      abort "Daemonization is not supported on Windows. Running 'start #{ChefZero::Dist::CLIENT}' will fork the process."
     else
       abort "Process.daemon requires Ruby >= 1.9"
```
Comment (tas50) [inline bin/chef-zero]: this bit makes me sad. We have some cleanup to do

**Comment by tas50** [comment]
@Tensibai can you make sure all the commits have the DCO sign off?

**Comment by Tensibai** [comment]
To be extra complete, changes in rspec.rb and socketless_server_map.rb does change the output text/exception text to what was coherent in my mind, there's a possibility this breaks someone tests out there.

**Comment by ramereth** [comment]
@tas50 any update on when this will get merged?

**Pull request merged**